lib/chitter_web.rb

```diff
@@ -0,0 +1,98 @@
+require 'sinatra/base'
+require 'rack-flash'
+require 'data_mapper'
+require 'tilt/erb'
+require_relative 'user'
+
+env = ENV['RACK_ENV'] || 'development'
+
+DataMapper.setup(:default, ENV['DATABASE_URL']  || "postgres://localhost/chitter_#{env}")
+
+require './lib/user'
+require './lib/peep' # this needs to be done after datamapper is initialised
+
+DataMapper.finalize
+
+DataMapper.auto_upgrade!
+
+class Chitter < Sinatra::Base
+
+	use Rack::Flash
+	use Rack::MethodOverride
+	enable :sessions
+	set :session_secret, 'extremely secret stuff'
+
+	helpers do
+		def current_user
+			@current_user ||= User.get(session[:user_id]) if session[:user_id]
```

Comment:
Is it necessary to define a `@current_user` instance variable here when it just gets returned? Could you not just do

```ruby
def current_user
  User.get(session[:user_id]) if session[:user_id]
end
```